### ISSUE TITLE
Unnecessary referenceName() allocation in AbstractReferenceCounted

### ISSUE THREAD

**Issue by glukos** [OpenHFT/Chronicle-Core#254]
```
        referenceCounted = (MonitorReferenceCounted) ReferenceCountedTracer.onReleased(performRelease, referenceName(), getClass());
```

```
    static ReferenceCountedTracer onReleased(final Runnable onRelease, String uniqueId, Class type) {
        return Jvm.isResourceTracing()
                ? new DualReferenceCounted(
                new TracingReferenceCounted(onRelease, uniqueId, type),
                new VanillaReferenceCounted(() -> {
                }, type))
                : new VanillaReferenceCounted(onRelease, type);
    }
```
ID is not used in case resource tracing is off, but we still call referenceName() and allocate a string

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.21.92](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.21.92), [BOM-2.21.240](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.21.240)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Core-2.22ea10](https://github.com/OpenHFT/Chronicle-Core/releases/chronicle-core-2.22ea10), [BOM-2.22ea40](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea40)
